k2/csrc/properties.cc

```diff
@@ -33,18 +33,16 @@ bool IsValid(const Fsa &fsa) {
     } else {
       // every state contains at least one arc.
       if (arc.src_state != state + 1) return false;
-      // `leaving_arcs` and `arcs` in this state are not consistent.
-      if ((fsa.leaving_arcs[state + 1].begin - fsa.leaving_arcs[state].begin) !=
-          num_arcs)
+      // `arc_indexes` and `arcs` in this state are not consistent.
+      if ((fsa.arc_indexes[state + 1] - fsa.arc_indexes[state]) != num_arcs)
         return false;
       state = arc.src_state;
       num_arcs = 1;
     }
   }
   // check the last state
   if (final_state != state + 1) return false;
-  if ((fsa.leaving_arcs[final_state].begin - fsa.leaving_arcs[state].begin) !=
-      num_arcs)
+  if ((fsa.arc_indexes[final_state] - fsa.arc_indexes[state]) != num_arcs)
```

Comment:
... this code is hard for me to understand and to verify that it's correct.  I think it would be much clearer if the iteration were done over fsa.arc_indexes.